packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -146,35 +167,88 @@ export class GCFBootstrapper {
     return config as Options;
   }
 
+  parseRequestHeaders(
+    request: express.Request
+  ): {name: string; id: string; signature: string; taskId: string} {
+    const name =
+      request.get('x-github-event') || request.get('X-GitHub-Event') || '';
+    const id =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    // TODO: add test to validate this signature is used on the initial
+    // webhook from GitHub:
+    const signature =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    const taskId =
+      request.get('X-CloudTasks-TaskName') ||
+      request.get('x-cloudtasks-taskname') ||
+      '';
+    return {name, id, signature, taskId};
+  }
+
+  parseTriggerType(name: string, taskId: string): TriggerType {
+    if (
+      !taskId &&
+      (name === 'schedule.repository' || name === 'pubsub.message')
+    ) {
+      return TriggerType.SCHEDULER;
+    } else if (!taskId && name) {
+      return TriggerType.GITHUB;
+    } else if (name) {
+      return TriggerType.TASK;
+    }
+    return TriggerType.UNKNOWN;
+  }
+
+  buildTriggerInfo(
+    triggerType: TriggerType,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestBody: {[key: string]: any}
```

Comment:
Need to have `any` here because the content of the payload depends on GitHub so can't really type-check it